### PR TITLE
Show original message in in-channel thread preview, aligned by author

### PR DESCRIPTION
**Pull request by lilseyi**
## What & why

In a group channel, when a message has thread replies the app floats a small preview near the bottom of the chat so recent thread activity isn't lost off-screen. Today that preview (`GhostThreadPointer`) is a **dashed, greyed, centered pill** showing only "N replies · Just now" — it doesn't show the original message, and because it's centered you can't tell who started the thread.

This makes two changes the contributor asked for:

1. **Show the original message** — the preview now echoes the original message's text (truncated to two lines).
2. **Align it like a real message** — **right (no avatar) when the current user authored** the original, **left (with the author's avatar + name) when someone else did**. Alignment keys off the **original message's author**, not the last replier, and reuses the same own/other bubble colors as normal messages.

The two tap targets are unchanged: tapping the "N replies" pill still opens the thread; tapping the bubble body still scrolls up to and highlights the real original message.

This deliberately reverses the component's original "content-free pointer, not a real message" design — its header comment is updated to match the new behavior.

## Rendered mock (not a device capture)

This runs on a headless Linux runner with no iOS simulator, so the image below is a **faithful rendered mock built from the real component's `StyleSheet` values and the light-theme tokens** — not a device screenshot. It covers both alignments plus the edge cases.

[After — rendered mock](https://images.togather.nyc/dev-assistant/8b8a9fdb-e4d8-4d89-a6db-253ad34ed8bd-ghost-after-mock.png)

## Edge cases handled

- **Long original** — truncates to two lines with an ellipsis.
- **Image-/attachment-only original** — shows a placeholder ("📷 Photo", "🎥 Video", "🎤 Voice message", "📄 <name>", "📎 Attachment") rather than a blank bubble.
- **Deleted original** — shows the deleted-message treatment (defensive; deleted originals don't normally float a ghost via `shouldFloatGhost`).
- **Missing avatar / long sender name** — falls back the same way normal rows do (initials placeholder, name truncated to one line).

## Changes

- `GhostThreadPointer.tsx` — echo the original message with own/other alignment + colors; carry `originalContent`/`originalSenderId`/`currentUserId`/sender info/`isDeleted`/`attachments`; keep both tap targets. Header comment rewritten.
- `MessageList.tsx` — forward the original message's text/author/sender info onto the `ghost` list item and into `GhostThreadPointer`.
- `EventActivity.tsx` — the event-activity feed's ghost usage gets the same new props.
- `__tests__/GhostThreadPointer.test.tsx` — flipped from asserting content-free to asserting the original text is shown and alignment follows authorship; added image-placeholder and deleted-message coverage. `threadTimeline` derivation is unchanged (it already carried the full message), so its tests stayed as-is.

## Verification

- `GhostThreadPointer` + `threadTimeline` unit tests pass.
- Full mobile Jest suite green (1273 passed, 9 skipped).
- `tsc` clean for the changed files; ESLint clean (only pre-existing `any` warnings in surrounding code).

Dashboard item: `x97fyhqp4ccdme5588vkt7gsvh8ac91n`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_01A8Yu7B3DEJaL6QaZxGEEqp)_

### CODE REVIEW

File: apps/mobile/features/chat/components/__tests__/GhostThreadPointer.test.tsx [hunk 1]
```diff
@@ -28,40 +45,109 @@ jest.mock('../ThreadReplies', () => {
...
+      />,
+    );
+
+    // Avatar + sender name are shown for other-authored previews (the mock's
+    // left-aligned treatment). The own-authored test asserts the inverse.
+    expect(getByText('Samuel Baker')).toBeTruthy();
+    expect(getByTestId('ghost-avatar')).toBeTruthy();
+  });
+
+  it('right-aligns with no avatar/name when the current user authored the original', () => {
```
Comment (lilseyi) [inline apps/mobile/features/chat/components/__tests__/GhostThreadPointer.test.tsx:96]: Non-blocking (test coverage). The suite was flipped correctly and now covers text-shown, avatar/name presence for other-authored, image-only placeholder, deleted, and both tap targets — good. Two small gaps worth closing while you're here:

- **Alignment is asserted only by proxy.** The own/other tests check avatar/name presence/absence but never assert the actual `justifyContent` (`flex-end` vs `flex-start`) on the row. Since alignment-by-author is the headline behavior of this change, an explicit style assertion would pin it down.
- **Truncation has no test.** `numberOfLines={2}` (`GhostThreadPointer.tsx:168`) is a "Done when" item in the spec but isn't exercised. A quick assert that the bubble `Text` renders with `numberOfLines={2}` for a long original would cover it.

Also minor: only the `image` attachment branch is tested — `video`/`audio`/`document`/generic and the empty-`"Message"` fallback in `getPreview` are untested, and the missing-`senderName` → `'Unknown'`/`'User'` fallback has no test. All low priority.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by lilseyi — COMMENTED** [review]
**Automated review (dev-assistant, medium-risk) — no blocking findings.**

Reviewed the diff against the approved spec across four lenses (correctness, security, spec-fidelity/UX, tests), and adversarially re-checked every candidate finding. Summary:

- **Correctness** — clean. `tsc --noEmit` passes for all three changed source files; the new `attachments`/`isDeleted`/`currentUserId` plumbing is type-safe in both consumers (`MessageList.tsx`, `EventActivity.tsx`).
- **Security** — no new exposure. The ghost echoes only data (`content`, `senderName`, `senderProfilePhoto`, `attachments`) that came from the same authorized `useMessages` result already rendered in the same view; content goes only into RN `<Text>`/`AppImage` (no injection surface).
- **Spec fidelity** — all 7 "Done when" items met: original text shown + truncated (`numberOfLines={2}`), alignment keys off the **original** author (`originalSenderId === currentUserId`) not the last replier, bubble colors match `MessageItem` tokens exactly, both tap targets preserved, edge cases (deleted/attachment-only/long/missing-avatar) handled, header comment rewritten. `EventActivity.tsx` is a forced signature-propagation on the shared component, not scope creep. No backend change.

**Non-blocking cleanup (worth a follow-up commit, not gating):**

1. **Stale "content-free" comments.** The spec asked to drop the old "content-free pointer" rationale — done in `GhostThreadPointer.tsx`, but the same now-contradictory wording still lives in three sibling spots that describe the ghost as "content-free":
   - `apps/mobile/features/chat/components/MessageList.tsx:254`
   - `apps/mobile/features/chat/utils/threadTimeline.ts:7`
   - `apps/mobile/features/chat/utils/threadTimeline.test.ts:9`
   
   Updating them to match the new echo behavior keeps the rationale consistent (per the same instruction that applied to the header comment).

2. **Test coverage gaps** — see the inline comment on the test file (alignment asserted only by proxy; `numberOfLines` truncation untested; the `MessageList` ghost-payload mapping has no test).

Verdict: **approved** — nothing here blocks merge; the items above are low-severity polish.

_Note: posted as a comment-level review; routine sessions can't submit a formal approve/request-changes review, so the approved verdict travels via the dashboard callback._

---
_Generated by [Claude Code](https://claude.ai/code)_
